sa/storage-authority.go

```diff
@@ -484,6 +548,33 @@ func (ssa *SQLStorageAuthority) FinalizeAuthorization(authz core.Authorization)
 		return
 	}
 
+	var challs []challModel
+	_, err = tx.Select(
+		&challs,
+		"SELECT * FROM challenges WHERE authorizationID = :authID ORDER BY id ASC",
+		map[string]interface{}{"authID": authz.ID},
+	)
+	if err != nil {
+		tx.Rollback()
+		return
+	}
+	if len(challs) != len(authz.Challenges) {
```

Comment:
@jsha opened an issue for this, the plan is to make these changes in another PR.